cpp/cmake_modules/FindabslAlt.cmake

```
@@ -0,0 +1,48 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+if(abslAlt_FOUND)
+  return()
+endif()
+
+set(find_package_args)
+
+if(abslAlt_FIND_VERSION)
+  list(APPEND find_package_args ${abslAlt_FIND_VERSION})
+endif()
+if(abslAlt_FIND_QUIETLY)
+  list(APPEND find_package_args QUIET)
+endif()
+
+if(abslAlt_FIND_REQUIRED)
+  list(APPEND find_package_args REQUIRED)
+endif()
+
+find_package(absl ${find_package_args})
+
+if(NOT DEFINED absl_VERSION)
+  # Abseil does not define a version when build 'live at head'.
+  # As this is their recommended path we need to define a large version to pass version checks.
+  set(absl_VERSION 99999999_head)
```

Do we need the `_head` part?

```suggestion
  set(absl_VERSION 99999999)
```

---

Technically not but the absl.pc uses 'head' as a version and it doesn't influence the comparision with our minum version and is shown in the configure log.

---

OK.

It seems that the `VERSION_*` operators remove the `_head` part:

https://cmake.org/cmake/help/latest/command/if.html#version-comparisons

> (version format is major[.minor[.patch[.tweak]]], omitted components are treated as zero). Any non-integer version component or non-integer trailing part of a version component effectively truncates the string at that point.

Could you add a comment that the `_head` part is safe for version comparison?